packages/ui/src/composables/useBem.ts

```diff
@@ -1,21 +1,72 @@
-import { computed, Ref, unref } from 'vue'
-
-export const useBem = (baseClass: string) => {
-  const createModifiersClasses = (modifiersCb: () => Record<string, boolean>, classes: string[] | Ref<string[]> = []) => {
-    return computed(() => {
-      const modifiers = modifiersCb()
-      return [
-        baseClass,
-        ...unref(classes),
-        ...Object
-          .keys(modifiers)
-          .filter((modifierName) => modifiers[modifierName])
-          .map((modifierName) => `${baseClass}--${modifierName}`),
-      ]
-    })
-  }
+import { computed, Ref, unref, ComputedRef } from 'vue'
+import { __DEV__ } from '../utils/global-utils'
+import isFunction from 'lodash/isFunction.js'
+
+type Key<Prefix extends string, ModifierKey extends string> = `${Prefix}--${ModifierKey}`
+
+type ClassesObject<Key extends string> = Record<Key, true>
+
+type ComputedClasses<Key extends string> = Ref<ClassesObject<Key>> & {
+  asObject: ComputedRef<ClassesObject<Key>>
+  asArray: ComputedRef<Key[]>
+  asString: ComputedRef<string>
+}
 
-  return {
-    createModifiersClasses,
+/**
+ * @description creates BEM modifiers based on transferred prefix (base BEM class) & modifiers list.
+ * @param prefix string that classes start with (base BEM class).
+ * @param modifiers list of options that will serve as state BEM modifiers.
+ * @returns computed classes starting with "prefix" and ending with form state BEM modifier.
+ * @example
+ *  const result = useBem('va-component', computed(() => pick(props, ['success'])))
+ *  // if success is `true`
+ *  { ...result }: { 'va-component--success': true }
+ *  result.value: { 'va-component--success': true }
+ *  result.asObject.value: { 'va-component--success': true }
+ *  result.asArray.value: ['va-component--success']
+ *  result.asString.value: 'va-component--success'
+ */
+export const useBem = <Prefix extends string, ModifierKey extends string>(
+  prefix: Prefix,
+  modifiers: Record<ModifierKey, boolean> | Ref<Record<ModifierKey, boolean>> | (() => Record<ModifierKey, boolean>),
+) => {
+  if (__DEV__ && !prefix) {
+    console.warn('You must pass the @param "prefix" to the useBem hook!')
   }
+
+  const modifiersList = isFunction(modifiers) ? modifiers() : modifiers
```

Comment:
modifiers will probably return non-reactive value. So it must be called in computed.

Anyway, it's better to create `computed(modfiers)` if modifiers is function, or `computed(() => unref(modifiers)...)` if it is an object.